doc/manual/compiler.rst

```diff
@@ -156,8 +156,12 @@ ARTIQ makes various useful built-in and mathematical functions from Python, NumP
 
 Basic NumPy array handling (``np.array()``, ``numpy.transpose()``, ``numpy.full``, ``@``, element-wise operation, etc.) is also available. NumPy functions are implicitly broadcast when applied to arrays. 
 
-.. warning:: 
-    A kernel ``print`` call normally specifically prints to the host machine, either the terminal of ``artiq_run`` or the dashboard log when using ``artiq_dashboard``. This makes it effectively an RPC, with some of the corresponding limitations. In subkernels and whenever compiled through ``artiq_compile``, where RPCs are not supported, it is instead considered a print to the local log (UART only in the case of satellites, UART and core log for master/standalone devices).    
+Print and logging functions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The built-in function ``print()`` is somewhat of a special case. Used in kernels, it normally functions as an RPC to the regular host Python ``print()``, i.e. with output either in the terminal of ``artiq_run`` or in the client logs when using ``artiq_dashboard`` or ``artiq_client``. Notably, it is not automatically asynchronous, and will affect ``rtio_clock_mu`` and the timing dynamics of surrounding statements accordingly. On the other hand, in subkernels and whenever compiled through ``artiq_compile``, where RPCs are not supported, it is instead considered a print to the core log (accessible by UART even in the absence of an Ethernet connection for ``artiq_coremgmt log``, as for example in satellites). 
```

Comment:
> it is not automatically asynchronous

I would either drop the "automatically" or elaborate on this further (how to make it asynchronous, then?).